### ISSUE TITLE
Error while monitoring ECS task due to missing log stream

### ISSUE THREAD

**Issue by madkinsz** [PrefectHQ/prefect-aws#177]
```
botocore.errorfactory.ResourceNotFoundException: An error occurred (ResourceNotFoundException)
when calling the GetLogEvents operation: The specified log stream does not exist.
```

![image](https://user-images.githubusercontent.com/2586601/207977581-1bf8e5bf-2eb6-455b-9ebe-42b75b1ea6c0.png)

Replicating a report from a user in the community: https://prefect-community.slack.com/archives/CL09KU1K7/p1671026163262599

I've seen a couple reports of this. Perhaps the log stream takes a bit to be populated? It'd be good to retry here. Unclear if this is an issue with the way the users are configuring their task definitions without more details.

**Comment by harry-hause** [comment]
Also seeing this issue when setting:

```
    stream_output=True,
    configure_cloudwatch_logs=True
```
in the ECSTask Block. 

Currently no other logging configuration. I've checked both versions 0.2.0 and 0.2.1.

**Comment by boseoladipo** [comment]
I also have this issue.

I'm trying to initialize an ECS task object in my flow, and immediately run the ecs task. I have image set to None and configure_cloudwatch_logs set to False to prevent prefect from registering a new task definition, but even with configure_cloudwatch_logs set to True I get the same error.
``` python
from prefect import flow, task, get_run_logger
from prefect_aws.ecs import ECSTask


@task
def ecs_flow_task():
    ecs_task = ECSTask(
        cluster="prefect-default-cluster",
        task_definition_arn="arn:aws:ecs:eu-central-1:*********:task-definition/default-adhoc-task",
        vpc_id="vpc-*********",
        stream_output=True,
        configure_cloudwatch_logs=False,
        image=None,
        task_customizations=[
            {
                "op": "add",
                "path": "/networkConfiguration/awsvpcConfiguration/subnets",
                "value": [
                    "subnet-*********",
                    "subnet-*********",
                    "subnet-*********",
                ],
            },
            {
                "op": "add",
                "path": "/networkConfiguration/awsvpcConfiguration/securityGroups",
                "value": ["sg-*********"],
            },
        ],
    )
    ecs_task.run()
    logger = get_run_logger()
    logger.info("Triggered ECS Task")


@flow()
def ecs_flow():
    ecs_flow_task()


if __name__ == "__main__":
    ecs_flow()

```

Here's the configuration of the task definition being passed in the fllow:
``` json
{
    "taskDefinitionArn": "arn:aws:ecs:eu-central-1:*********:task-definition/default-adhoc-task:5",
    "containerDefinitions": [
        {
            "name": "prefect",
            "image": "*********.dkr.ecr.eu-central-1.amazonaws.com/adhoc-task",
            "cpu": 64,
            "memory": 256,
            "memoryReservation": 128,
            "portMappings": [],
            "essential": true,
            "environment": [],
            "mountPoints": [],
            "volumesFrom": [],
            "logConfiguration": {
                "logDriver": "awslogs",
                "options": {
                    "awslogs-group": "prefect-default-agent-logs",
                    "awslogs-region": "eu-central-1",
                    "awslogs-stream-prefix": "adhoc-task-definition"
                }
            }
        }
    ],
    "family": "default-adhoc-task",
    "taskRoleArn": "arn:aws:iam::*********:role/prefect-default-ECSTaskRole",
    "executionRoleArn": "arn:aws:iam::*********:role/prefect-default-ECSExecutionTaskRole",
    "networkMode": "awsvpc",
    "revision": 5,
    "volumes": [],
    "status": "INACTIVE",
    "requiresAttributes": [
        {
            "name": "com.amazonaws.ecs.capability.logging-driver.awslogs"
        },
        {
            "name": "ecs.capability.execution-role-awslogs"
        },
        {
            "name": "com.amazonaws.ecs.capability.ecr-auth"
        },
        {
            "name": "com.amazonaws.ecs.capability.docker-remote-api.1.19"
        },
        {
            "name": "com.amazonaws.ecs.capability.docker-remote-api.1.21"
        },
        {
            "name": "com.amazonaws.ecs.capability.task-iam-role"
        },
        {
            "name": "ecs.capability.execution-role-ecr-pull"
        },
        {
            "name": "com.amazonaws.ecs.capability.docker-remote-api.1.18"
        },
        {
            "name": "ecs.capability.task-eni"
        }
    ],
    "placementConstraints": [],
    "compatibilities": [
        "EC2",
        "FARGATE"
    ],
    "requiresCompatibilities": [
        "FARGATE"
    ],
    "cpu": "512",
    "memory": "1024",
    "registeredAt": "2022-12-30T21:25:43.736Z",
    "deregisteredAt": "2023-01-02T12:34:13.559Z",
    "registeredBy": "arn:aws:sts::*******:assumed-role/terraformExecutionRole/aws-go-sdk-1672435543204924000",
    "tags": []
}
```

I deploy the flow using a different ECS task definition, saved as an ECS task block in Prefect Cloud. I can see the logs created by the task run, but prefect seems to be trying to read from a log stream that doesn't exist. As far as I can tell the full log stream for task definition run cannot be manipulated by the user, and matches what AWS specified in the documentation.

Is there a basis for prefect trying to read from [this log stream](https://github.com/PrefectHQ/prefect-aws/blob/267d973fcf9f9ccf1a4508140daf5b033a53c609/prefect_aws/ecs.py#L964-L970):
```
stream_name = "/".join(
                    [
                        log_config["awslogs-stream-prefix"],
                        task_definition["family"],
                        task_arn.rsplit("/")[-1],
                    ]
                )
```
Instead of [this](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_awslogs.html):
![image](https://user-images.githubusercontent.com/40306122/210554210-84714541-22c5-4713-97ee-d28a7fb6df86.png)

**Comment by boseoladipo** [comment]
Thanks for getting this fixed! I guess I'll be able to test this once the fix is merged.

**Comment by madkinsz** [comment]
@boseoladipo np! fyi you can always install from a git branch i.e. `pip install git+https://github.com/PrefectHQ/prefect-aws@fix/ecs-log-stream`.
